src/qutip_cupy/__init__.py

```diff
@@ -1 +1,25 @@
 from .version import version as __version__
+from .cupy_dense import *
```

Comment:
I would not use `import *` here because it hinders static analysis tools like `pyflakes`. I would do something like:
```python
import .cupy_dense as cd

CuPyDense = cd.CuPyDense

...

del cd
del data
```

Reply:
How about just `from .cupy_dense import CuPyDense`?